lib/matplotlib/backends/backend_qt5.py

```diff
@@ -1019,6 +1031,80 @@ def trigger(self, *args, **kwargs):
 backend_tools.ToolCopyToClipboard = ToolCopyToClipboardQT
 
 
+@cbook.deprecated("3.0")
+def error_msg_qt(msg, parent=None):
+    if not isinstance(msg, str):
+        msg = ','.join(map(str, msg))
+
+    QtWidgets.QMessageBox.warning(None, "Matplotlib",
+                                  msg, QtGui.QMessageBox.Ok)
+
+
+@cbook.deprecated("3.0")
+def exception_handler(type, value, tb):
+    """Handle uncaught exceptions
+    It does not catch SystemExit
+    """
+    msg = ''
+    # get the filename attribute if available (for IOError)
+    if hasattr(value, 'filename') and value.filename is not None:
+        msg = value.filename + ': '
+    if hasattr(value, 'strerror') and value.strerror is not None:
+        msg += value.strerror
+    else:
+        msg += str(value)
+
+    if len(msg):
+        error_msg_qt(msg)
+
+class allow_interrupt():
+
+    def __init__(self, interrupted_qobject):
+        QAS = QtNetwork.QAbstractSocket
+        self.qt_socket = QAS(QAS.TcpSocket, qApp)
+        self.old_fd = None
+        self.old_sigint_handler = None
+        self.interrupt_caught = False
+        self.interrupted_qobject = interrupted_qobject
+        # Create a socket pair
+        self.wsock, self.rsock = socket.socketpair()
+        # Let Qt listen on the one end
+        self.qt_socket.setSocketDescriptor(self.rsock.fileno())
+        self.wsock.setblocking(False)
+        # First Python code executed gets any exception from
+        # the signal handler, so add a dummy handler first
+        self.qt_socket.readyRead.connect(lambda: None)
+        # Second handler does the real handling
+        self.qt_socket.readyRead.connect(self._readSignal)
+
+    def __enter__(self):
+        # And let Python write on the other end
+        self.old_fd = signal.set_wakeup_fd(self.wsock.fileno())
+        self.old_sigint_handler = signal.getsignal(signal.SIGINT)
+        signal.signal(signal.SIGINT, self._handle)
+
+    def __exit__(self, type, val, traceback):
+        signal.set_wakeup_fd(self.old_fd)
+        signal.signal(signal.SIGINT, self.old_sigint_handler)
+        self.wsock.close()
+        self.rsock.close()
+        self.qt_socket.abort()
+        if self.interrupt_caught:
+            self.old_sigint_handler()
+
+    def _readSignal(self):
+        # Read the written byte.
+        # Note: readyRead is blocked from occurring again until readData()
+        # was called, so call it, even if you don't need the value.
+        data = self.qt_socket.readData(1)
+
+    def _handle(self, *args):
+        self.interrupt_caught = True
+        if self.interrupted_qobject is qApp:
+            pyplot.close()
```

Comment:
I am not sure we want this.  This will only close the current figure (not all of them).  We should either just leave the windows be (assuming that something outside of us will spin the event loop or not as they want) or do `plt.close('all')` 

I strongly prefer we leave the figures alone here.